src/server/decode.rs

```diff
@@ -4,12 +4,13 @@ use std::str::FromStr;
 
 use async_std::io::{BufReader, Read, Write};
 use async_std::prelude::*;
-use http_types::headers::{CONTENT_LENGTH, EXPECT, HOST, TRANSFER_ENCODING};
+use http_types::headers::{CONTENT_LENGTH, EXPECT, TRANSFER_ENCODING};
 use http_types::{ensure, ensure_eq, format_err};
 use http_types::{Body, Method, Request};
 
 use crate::chunked::ChunkedDecoder;
 use crate::{MAX_HEADERS, MAX_HEAD_LENGTH};
+use url::Url;
```

Comment:
we can use `http_types::Url` here instead.

Reply:
addressed in 65d593a (and also dropped url from the crate because there was only one other usage, slight scope creep for this pr)